web/crux/src/app/version/version.service.ts

```diff
@@ -125,8 +141,47 @@ export default class VersionService {
               },
             }),
         )
-
         await Promise.all(images)
+
+        await Promise.all(
```

Comment:
could you map these to a local variable like `images` are and add the two `await Promise.all()` call at the end of the function please? That way they would utilize the db more efficient i believe